phpBB/phpbb/files/filespec.php

```diff
@@ -0,0 +1,584 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\files;
+
+use phpbb\language\language;
+
+/**
+ * Responsible for holding all file relevant information, as well as doing file-specific operations.
+ * The {@link fileupload fileupload class} can be used to upload several files, each of them being this object to operate further on.
+ */
+class filespec
+{
+	/** @var string File name */
+	protected $filename = '';
+
+	/** @var string Real name of file */
+	protected $realname = '';
+
+	/** @var string Upload name of file */
+	protected $uploadname = '';
+
+	/** @var string Mimetype of file */
+	protected $mimetype = '';
+
+	/** @var string File extension */
+	public $extension = '';
```

Comment:
All public class members are accessed publicly by other classes/functions. I've only set them to public if they were being used externally. I'll check again if they might be able to just use the filespec->get() method instead.